sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFEvaluators.scala

```diff
@@ -129,7 +129,7 @@ class HiveGenericUDFEvaluator(
   override def returnType: DataType = inspectorToDataType(returnInspector)
 
   def setArg(index: Int, arg: Any): Unit =
-    deferredObjects(index).asInstanceOf[DeferredObjectAdapter].set(arg)
+    deferredObjects(index).asInstanceOf[DeferredObjectAdapter].set(() => arg)
```

Comment:
cc @jackylee-ch Can you test this fix?

Reply:
> cc @jackylee-ch Can you test this fix?

I have some change in my local. Like @cloud-fan said, we should pass function to setArg to make it lazy. I have passed the failed case with none codegen case, and I'm still trying to fix it with codegen.

Reply:
@jackylee-ch Can you help test the updated version again? 
Thanks.